src/js/components/List/List.js

```diff
@@ -105,32 +148,74 @@ const List = React.forwardRef(
     const Container = paginate ? StyledContainer : Fragment;
     const containterProps = paginate ? { ...theme.list.container } : undefined;
 
+    const [orderingData, setOrderingData] = useState();
+
+    const draggingRef = useRef();
+    
+    const ariaProps = {
+      role: onClickItem || onOrder ? 'listbox' : 'list',
+    };
+    
+    
+    if (active >= 0) {
+      let activeId;
+      // We have an item that is 'focused' within the list. This could
+      // be the list item or one of the up/down ordering buttons.
+      // We need to figure out an id of the thing that will be shown as active
+      if (onOrder) {
+        // figure out which arrow button will be the active one.
+        const btnId = (active % 2) ? 'MoveDown' : 'MoveUp';
+        const itemIndex = Math.trunc(active / 2);
+        activeId = 
+          `${chooseItemId(data[itemIndex], itemIndex, primaryKey)}${btnId}`;
+      } else if (onClickItem) {
+        // The whole list item is active. Figure out an id
+        activeId = chooseItemId(data[active], active, primaryKey);
+      }
+      ariaProps['aria-activedescendant'] = activeId;
+    }
+
     return (
       <Container {...containterProps}>
         <Keyboard
           onEnter={
-            onClickItem && active >= 0
+            (onClickItem || onOrder) && active >= 0
               ? event => {
-                  event.persist();
-                  const adjustedEvent = event;
-                  adjustedEvent.item = data[active];
-                  adjustedEvent.index = active;
-                  onClickItem(adjustedEvent);
+                  if (onOrder) {
+                    const index = Math.trunc(active / 2);
+                    if (active % 2) {
+                      onOrder(reorder(data, index, index + 1));
+                      setActive(Math.min(active + 2, data.length * 2 - 2));
```

Comment:
Maybe a comment here explaining that we are adjusting `active` so that the same control on the moved item remains active after the reorder?